### PR TITLE
add support for migration dir format

### PR DESCRIPTION
**Pull request by zeevmoney**

### CODE REVIEW

File: __tests__/atlas.test.ts [hunk 1]
```diff
@@ -133,6 +133,36 @@ describe('run with latest', () => {
...
+    expect(res.fileReports?.[1].Error).toBeFalsy()
+  })
+
+  test('fail on wrong migration dir format', async () => {
+    // Actions creates an environment variables for inputs (in action.yaml), syntax: INPUT_<VARIABLE_NAME>.
+    process.env['INPUT_DIR-FORMAT'] = 'incorrect'
+    const res = (await run()) as AtlasResult
+    expect(res.exitCode).toEqual(ExitCodes.Failure)
+    expect(res.raw.trimEnd()).toEqual(
+      'Error: sql/migrate: stat : no such file or directory'
```
Comment (zeevmoney) [inline __tests__/atlas.test.ts:163]: this test has a wrong err msg, fixing it, fixing it.

File: __tests__/atlas.test.ts [hunk 2]
```diff
@@ -133,6 +133,36 @@ describe('run with latest', () => {
...
+    // Actions creates an environment variables for inputs (in action.yaml), syntax: INPUT_<VARIABLE_NAME>.
+    process.env.INPUT_DIR = path.join('__tests__', 'testdata', 'golang-migrate')
+    process.env['INPUT_DIR-FORMAT'] = 'golang-migrate'
+    process.env['INPUT_LATEST'] = '4'
+    const res = (await run()) as AtlasResult
+    expect(res.exitCode).toEqual(ExitCodes.Success)
+    expect(res.fileReports).toHaveLength(2)
+    expect(res.fileReports?.[0].Name).toEqual('1_initial.up.sql')
+    expect(res.fileReports?.[0].Text.replace(/\s\s+|\n/g, ' ')).toEqual(
+      'CREATE TABLE tbl ( col INT );'
```
Comment (yonidavidson) [inline __tests__/atlas.test.ts:147]: Why do you need this? I think its hard to maintain

File: __tests__/atlas.test.ts [hunk 3]
```diff
@@ -133,6 +133,36 @@ describe('run with latest', () => {
...
+    )
+    expect(res.fileReports?.[0].Error).toBeFalsy()
+    expect(res.fileReports?.[1].Name).toEqual('2_second_migration.up.sql')
+    expect(res.fileReports?.[1].Text.replace(/\s\s+|\n/g, ' ')).toEqual(
+      'CREATE TABLE tbl_2 (col INT);'
```
Comment (yonidavidson) [inline __tests__/atlas.test.ts:152]: here too

File: __tests__/atlas.test.ts (hunk 2 above)
Comment (zeevmoney) [reply __tests__/atlas.test.ts]: replaced

File: __tests__/atlas.test.ts (hunk 3 above)
Comment (zeevmoney) [reply __tests__/atlas.test.ts]: replaced

File: action.yml [hunk 4]
```diff
@@ -25,7 +25,10 @@ inputs:
   ariga-token:
     description: 'Token for sending the report to the cloud (should be passed as a secret)'
     required: false
-
+  dir-format:
```
Comment (a8m) [inline action.yml:28]: Keep it close to the `dir`.

Comment (a8m) [reply action.yml:28]: Also, use `dev-url` to follow the Atlas names.

File: __tests__/atlas.test.ts [hunk 5]
```diff
@@ -4,12 +4,40 @@ import { tmpdir } from 'os'
...
+} from '../src/cloud'
+
+jest.mock('../src/cloud', () => {
+  const actual = jest.requireActual('../src/cloud')
+  return {
+    ...actual,
+    getDownloadURL: jest.fn(
+      (version: string) =>
+        new URL(
+          `${BASE_ADDRESS}/${S3_FOLDER}/atlas-${ARCHITECTURE}-${version}?test=1`
```
Comment (yonidavidson) [inline __tests__/atlas.test.ts:36]: just saying, you could have a helper function that adds the test=1 query param, since it's used multiple times

**Review by yonidavidson — APPROVED** [review]

File: __tests__/atlas.test.ts (hunk 5 above)
Comment (zeevmoney) [reply __tests__/atlas.test.ts:36]: It means that I will have to check if I'm inside a test inside the actual code, really ugly.

**Pull request merged**